src/UnoFlipController.java

```diff
@@ -86,42 +105,62 @@ public void actionPerformed(ActionEvent e) {
                         null,
                         null);
 
-                Card.Colour colour;
-                if (result == JOptionPane.OK_OPTION) {
+                // Colour is selected and OK is pressed
+                if (result == JOptionPane.OK_OPTION)
+                {
+                    // Colour to set the new current colour of the game
+                    Card.Colour colour;
+                    // Parse the input given by the user from the drop-down to set the new colour
                     String option = (String) comboBox.getSelectedItem();
                     colour = Card.Colour.valueOf(option);
+                    // Sets the new current colour
                     this.model.setCurrentColour(colour);
+                    // Notify view to show status of the current colour
                     this.model.notifyViews();
                     break;
-                } else if (result == JOptionPane.CLOSED_OPTION){
-                    continue;
                 }
             }
 
-        } else if(e.getActionCommand().equals(UnoFlipViewFrame.DRAW_CMD)){
+        }
+        // Draw card command for the player to draw a card
+        else if(e.getActionCommand().equals(UnoFlipViewFrame.DRAW_CMD))
+        {
+            // Model playTurn is called with -1 to signify a self draw one
             this.model.playTurn(-1);
         }
-        else if(e.getActionCommand().equals(UnoFlipViewFrame.NEXT_CMD)) {
+        // Next turn command to go to the next player
+        else if(e.getActionCommand().equals(UnoFlipViewFrame.NEXT_CMD))
+        {
+            // Model nextTurn is called
             this.model.nextTurn();
-        } else if (e.getActionCommand().equals(UnoFlipViewFrame.CHALLENGE_CMD) ) {
-
-            boolean challenge;
+        }
+        // Challenge command for wild draw 2 cards, prompts user if they want to challenge
+        else if (e.getActionCommand().equals(UnoFlipViewFrame.CHALLENGE_CMD) )
+        {
+            // Result from JOptionPane, Yes or No.
             int result = JOptionPane.showConfirmDialog(null, "Do you want to continue?", "Confirmation", JOptionPane.YES_NO_OPTION);
 
-            if (result == JOptionPane.YES_OPTION) {
-                challenge = true;
-            } else {
-                challenge = false;
-            }
+            // Challenge variable to know if the user wants challenge or not, true if Yes, otherwise false
+            boolean challenge = result == JOptionPane.YES_OPTION;
+            // Set the challenge variable in model
             this.model.setChallenge(challenge);
+            // Set the dontAsk variable to true for game not to prompt user again for the challenge
             this.model.setDontAsk(true);
```

Comment:
Why should the game prompt the user again for the challenge? This probably needs a different solution, we'll likely come back this in a separate issue in the near future.